stubs/Flask-Migrate/flask_migrate/__init__.pyi

```diff
@@ -0,0 +1,105 @@
+from collections.abc import Callable
+from logging import Logger
+from typing import Any, TypeVar
+from typing_extensions import ParamSpec, TypeAlias
+
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import MetaData
+
+_T = TypeVar("_T")
+_P = ParamSpec("_P")
+_App: TypeAlias = Any  # flask.Flask is not possible as a dependency yet
+
+alembic_version: tuple[int, int, int]
+log: Logger
+
+class _MigrateConfig:
+    migrate: Migrate
+    db: SQLAlchemy | None
+    directory: str
+    configure_args: dict[str, Any]
+    def __init__(self, migrate: Migrate, db: SQLAlchemy | None, **kwargs) -> None: ...
+    @property
+    def metadata(self) -> MetaData: ...
+
+class Config:  # should inherit from alembic.config.Config which is not possible yet
+    template_directory: str | None
+    def __init__(self, *args, **kwargs) -> None: ...
+    def get_template_directory(self) -> str: ...
+
+class Migrate:
+    configure_callbacks: list[Callable[[Config], None]]
```

Comment:
I'm not sure it's true that "configure callbacks" have to return `None`. That would make these functions a little odd, which return whatever the callback functions return:

- https://github.com/miguelgrinberg/Flask-Migrate/blob/7cb4236327ea04fc6be8a17bbfadae6de7bfbc8b/src/flask_migrate/__init__.py#L79-L82
- https://github.com/miguelgrinberg/Flask-Migrate/blob/7cb4236327ea04fc6be8a17bbfadae6de7bfbc8b/src/flask_migrate/__init__.py#L104

I'm not sure what they _should_ return, so (unless you have a better idea than me), maybe we should go with `Callable[[Config], Incomplete]` for now.

Also: It would be good to define a `_ConfigureCallback: TypeAlias = Callable[[Config], Incomplete]` alias somewhere, so that the type isn't repeated numerous times across the file.